### PR TITLE
Replace lint-license.sh with make lint-license

### PR DESCRIPTION
**Pull request by karlkfi**
Calling go-licenses from inside the makefile makes it easier to ensure that we are installing and using the vendored version every time.

This also fixes an issue in Go 1.22 that was causing errors related to missing module info.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -355,8 +358,21 @@ lint-bash:
 	@./scripts/lint-bash.sh
 
 .PHONY: lint-license
-lint-license: buildenv-dirs
-	./scripts/lint-license.sh
```
Comment (nan-yu) [inline Makefile:359]: `lint-license.sh` can be removed.

Comment (karlkfi) [reply Makefile:359]: oops. thought i had done that. fixed.

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**